### PR TITLE
feat: redesign Workflow Results 

### PR DESCRIPTION
**Pull request by mareklibra**
[FLPATH-1326](https://issues.redhat.com//browse/FLPATH-1326)

Related to:
- https://github.com/parodos-dev/backstage-orchestrator-workflows/pull/2
- https://github.com/parodos-dev/serverless-workflows/pull/403

TODO:
- [x] Try flat structure of workflows/Output
- [x] Fix Details' status - to be aligned with results. Introduce `Completed with Error`

### CODE REVIEW

**Comment by mareklibra** [comment]
Screenshot from 2024-09-23 14-46-35

---

Screenshot from 2024-09-23 14-46-51

---

Screenshot from 2024-09-23 14-46-58

---

Screenshot from 2024-09-23 14-47-20

File: plugins/orchestrator/src/components/WorkflowResult.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  error?: ProcessInstanceErrorDTO;
+  resultMessage?: WorkflowResultDTO['message'];
+  completedWith: WorkflowResultDTO['completedWith'];
+}) => {
+  const styles = useStyles();
+
+  const errorMessage = error?.message || error?.toString();
+  let noResult = <></>;
+  if (!resultMessage && !errorMessage) {
+    if (['Error', 'Completed', 'Aborted', 'Suspended'].includes(status || '')) {
```
[the remark this answers is not in the document]
Comment (batzionb) [reply plugins/orchestrator/src/components/WorkflowResult.tsx:62]: yeah better use the enum ProcessInstanceStatusDTO

File: plugins/orchestrator/src/components/WorkflowInstancePageContent.tsx [hunk 2]
```diff
@@ -269,7 +130,8 @@ export const WorkflowInstancePageContent: React.FC<{
           </InfoCard>
         </Grid>
 
-        {nextWorkflows.length > 0 ? (
+        {/* TODO: remove following, FLPATH-1672 */}
```
[the remark this answers is not in the document]
Comment (batzionb) [reply plugins/orchestrator/src/components/WorkflowInstancePageContent.tsx:136]: maybe just remove the comment?
we have the history in git...

**Comment by batzionb** [comment]
really small nit on UX
Can you make the error icon align to center with the error message?

**Comment by batzionb** [comment]
another small nit - 
I think the extra left margin before the links makes it look inconsistent

**Comment by batzionb** [comment]
Looks really good!

File: plugins/orchestrator-common/src/openapi/openapi.yaml [hunk 3]
```diff
@@ -493,22 +493,61 @@ components:
...
+        result:
+          $ref: '#/components/schemas/WorkflowResultDTO'
       additionalProperties: true
-    WorkflowOptionsDTO:
-      type: array
-      items:
-        $ref: '#/components/schemas/WorkflowSuggestionDTO'
-    WorkflowSuggestionDTO:
+    WorkflowResultDTO:
+      # Keep in sync with https://github.com/parodos-dev/serverless-workflows/blob/main/shared/schemas/workflow-result-schema.json
```
[the remark this answers is not in the document]
Comment (mareklibra) [reply plugins/orchestrator-common/src/openapi/openapi.yaml:500]: yep, just a left over

File: plugins/orchestrator/src/components/WorkflowInstancePageContent.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mareklibra) [reply plugins/orchestrator/src/components/WorkflowInstancePageContent.tsx:136]: yep, just a left-over

File: plugins/orchestrator-common/src/openapi/openapi.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mareklibra) [reply plugins/orchestrator-common/src/openapi/openapi.yaml:500]: We should help future maintainers to keep both sides of the contract in sync.
What about rewording the comment?

I would love to place similar comment into serverless-workflows, but it's json. Maybe to the documentation - will search for it

[the remark this answers is not in the document]
Comment (batzionb) [reply plugins/orchestrator-common/src/openapi/openapi.yaml]: @gciavarrini 
WDYM by independent? 
Is it OK now?

**Review by batzionb — APPROVED** [review]

**Comment by batzionb** [comment]
@gciavarrini 
If PR ok with you add /lgtm and merge it

File: plugins/orchestrator-common/src/openapi/openapi.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gciavarrini) [reply plugins/orchestrator-common/src/openapi/openapi.yaml]: @batzionb 
openapi spec contains the endpoint definitions. The definitions can't be dependent on the implementation of other services/code that uses the endpoints, it should be the other way around. That's what I mean

@mareklibra i see your point. Thanks for clarifying

**Review by batzionb — APPROVED** [review]

**Pull request merged**